flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MySqlToPaimonE2eITCase.java

```diff
@@ -0,0 +1,371 @@
+                                RowKind.INSERT,
+                                107,
+                                "rocks",
+                                "box of assorted rocks",
+                                5.1f,
+                                null,
+                                null),
+                        Row.ofKind(
+                                RowKind.INSERT,
+                                108,
+                                "jacket",
+                                "water resistent black wind breaker",
+                                0.1f,
+                                null,
+                                null),
+                        Row.ofKind(
+                                RowKind.INSERT,
+                                109,
+                                "spare tire",
+                                "24 inch spare tire",
+                                22.2f,
+                                null,
+                                null),
+                        Row.ofKind(
+                                RowKind.INSERT,
+                                110,
+                                "jacket",
+                                "water resistent white wind breaker",
+                                0.2f,
+                                null,
+                                null),
+                        Row.ofKind(
+                                RowKind.INSERT,
+                                111,
+                                "scooter",
+                                "Big 2-wheel scooter ",
+                                5.18f,
+                                null,
+                                null),
+                        Row.ofKind(RowKind.INSERT, 112, "finally", null, 2.14f, null, null)));
+    }
+
+    private void validateSinkResult(String query, List<Row> expected) {
+        long startWaitingTimestamp = System.currentTimeMillis();
+        boolean validateSucceed = false;
+        while (!validateSucceed) {
+            try {
+                List<Row> results = new ArrayList<>();
+                tEnv.executeSql(query).collect().forEachRemaining(results::add);
+                assertEqualsInAnyOrder(expected, results);
+                validateSucceed = true;
+            } catch (Throwable e) {
+                if (System.currentTimeMillis() - startWaitingTimestamp
+                        > TESTCASE_TIMEOUT_SECONDS * 1000_1000L) {
+                    throw new RuntimeException("Failed to check result with given query.");
+                }
+            }
+        }
+    }
+
+    private static void assertEqualsInAnyOrder(List<Row> expected, List<Row> actual) {
+        assertTrue(expected != null && actual != null);
+        assertEquals(expected.size(), actual.size());
+        Set<Row> expectedSet = new HashSet<>(expected);
+        for (Row row : actual) {
+            Assert.assertTrue(expectedSet.contains(row));
+        }
+    }
+}
```

Comment:
This `assertEqualsInAnyOrder` is a little suspicious since calling it with `expected: [1, 2, 3], actual: [1, 1, 2]` returns true. Perhaps snippets in other test cases like

```java
    public static void assertEqualsInAnyOrder(List<String> expected, List<String> actual) {
        assertTrue(expected != null && actual != null);
        assertEqualsInOrder(
                expected.stream().sorted().collect(Collectors.toList()),
                actual.stream().sorted().collect(Collectors.toList()));
    }
```
is fine.

Reply:
Actually, row is not implemented Comparable, so it can be used in sorted method.
I've added 'order by' query and asserted the result in same order.